File: JSON-Parsing-Example-2 (ObjC)/JSON Parsing Example 2/ViewController.m
//
//  ViewController.m
//  JSON Parsing Example 2
//
//  Created by Mallikarjun on 11/10/18.
//  Copyright © 2018 Mallikarjun. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
}


- (IBAction)buttonClicked:(id)sender {
    
    NSString *sampleUrl = @"http://182.72.79.154/iphonetest/getTheData.php";
    
    NSCharacterSet *set= [NSCharacterSet URLQueryAllowedCharacterSet];
    
    NSString * encodedURL = [sampleUrl stringByAddingPercentEncodingWithAllowedCharacters:set];
    
    NSURLSession * session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]];
    
    [[session dataTaskWithURL:[NSURL URLWithString:encodedURL] completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
        
        NSError * jsonError;
        
        id jsonResponse = [NSJSONSerialization JSONObjectWithData:data options:0 error:&jsonError];
        NSLog(@"JSON Response is: %@",jsonResponse);
        
         dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_LOW, 0), ^{
            
            dispatch_async(dispatch_get_main_queue(), ^{
                
                self->_textViewInstance.text = [NSString stringWithFormat:@"%@",jsonResponse];
                
            });
            
        });
        
    }]resume];
    
}
@end

File: Post-API-Call-using-Swift-3/Post API Call using Swift 3/ViewController.swift
//
//  ViewController.swift
//  Post API Call using Swift 3
//
//  Created by Mallikarjun on 27/11/18.
//  Copyright © 2018 Ladybird Web Solution. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    
    @IBOutlet weak var textViewToShowResult: UITextView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    
    @IBAction func postAPIMethodCalled(_ sender: Any) {
        
        
        let parameters = ["username" : "demoadmin", "password": "demopass"];
        
        guard let url = URL(string: "http://productdemourl.com/servicedesk38t/public/api/v1/authenticate") else { return}
        
        var request = URLRequest(url: url)
        
        request.httpMethod = "POST"
        request.addValue("application/json", forHTTPHeaderField: "content-type")
       
        guard let httpBody = try? JSONSerialization.data(withJSONObject: parameters, options: []) else { return }
        
        request.httpBody = httpBody
        
        let session = URLSession.shared
        
        
        
        session.dataTask(with: request) { (data, response, error) in
            
            if let response = response{
                print(response)
            }
            
            if let data = data{
                
                do{
                    
                   // let json = try JSONSerialization.jsonObject(with: data, options: [])
                    let json = try JSONSerialization.jsonObject(with: data, options: .allowFragments)
                    print(json)
                    
                } catch{
                    print(error)
                }
            }
            
            
        }.resume()
        
        
    }
    


}

